ios/FluentUI/People Picker/ContactCollectionView.swift

```diff
@@ -0,0 +1,161 @@
+
+@objc(MSFContactCollectionView)
+open class ContactCollectionView: UICollectionView {
+
+    /// Initializes the collection view by setting the layout, constraints, and cell to be used.
+    ///
+    /// - Parameters:
+    ///   - personaData: Array of PersonaData used to create each individual ContactView
+    @objc public init(personaData: [PersonaData] = []) {
+        layout = ContactCollectionViewLayout()
+        layout.scrollDirection = .horizontal
+        super.init(frame: .zero, collectionViewLayout: layout)
+
+        if personaData.count > 0 {
+            contactList = personaData
+        }
+
+        configureCollectionView()
+        setupHeightConstraint()
+        register(ContactCollectionViewCell.self, forCellWithReuseIdentifier: ContactCollectionViewCell.identifier)
+        NotificationCenter.default.addObserver(self, selector: #selector(setupHeightConstraint), name: UIContentSizeCategory.didChangeNotification, object: nil)
+    }
+
+    @objc public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    /// The array of PersonaData which is used to create each ContactView.
+    /// The height constraint of the ContactCollectionView is updated when the count increases from 0 or decreases to 0.
+    @objc public var contactList: [PersonaData] = [] {
+        didSet {
+            if (oldValue.count == 0 && contactList.count > 0) || (oldValue.count > 0 && contactList.count == 0) {
+                setupHeightConstraint()
+            }
+        }
+    }
+
+    open weak var contactCollectionViewDelegate: ContactCollectionViewDelegate?
+
+    private func configureCollectionView() {
+        translatesAutoresizingMaskIntoConstraints = false
+        showsHorizontalScrollIndicator = false
+        showsVerticalScrollIndicator = false
+        backgroundColor = Colors.surfacePrimary
+        dataSource = self
+        delegate = self
+        contentInset = UIEdgeInsets(top: 0, left: Constants.leadingInset, bottom: 0, right: 0)
+    }
+
+    @objc private func setupHeightConstraint() {
+        let height = (contactList.count > 0) ? UIApplication.shared.preferredContentSizeCategory.contactHeight : 0
+        heightConstraint.constant = height
+    }
+
+    private struct Constants {
+        static let leadingInset: CGFloat = 16.0
+    }
+
+    private let layout: ContactCollectionViewLayout
+    private var widthConstraint: NSLayoutConstraint?
+    private lazy var heightConstraint: NSLayoutConstraint = {
+        let heightConstraint = heightAnchor.constraint(equalToConstant: 0.0)
+        heightConstraint.isActive = true
+        return heightConstraint
+    }()
+    private var currentTappedIndex: Int?
+}
+
+extension ContactCollectionView: ContactViewDelegate {
+    public func didTapContactView(_ contact: ContactView) {
+        if let contactCollectionViewDelegate = contactCollectionViewDelegate, let currentTappedIndex = currentTappedIndex {
```

Comment:
There must be a better way calculate the tapped index from the ContactView reference.
You should get rid of the currentTappedIndex property. There should only be once source of truth for the tapped contact view. Currently, you are mixing the ContactView's tap handling and the UICollectionView's tap callback.

Reply:
That was what I originally was thinking. After some more thought about it, I don't think there is a way to calculate the tapped index from the `ContactView` reference because a `ContactView` shouldn't have a concept of an index since it doesn't have to be placed inside  a `ContactCollectionView`.

Reply:
Ah, I forgot that there isn't a way to just get a list of collection view cells. Ignore my comment, this'll work.